File: app/src/main/scala/lesson8/android/weather/DatabaseHelper.scala
package lesson8.android.weather

import android.content.Context
import android.database.sqlite.{SQLiteDatabase, SQLiteOpenHelper}
import android.provider.BaseColumns

object DatabaseHelper {
  private val DATABASE_VERSION: Int = 1
  val DATABASE_NAME = "weather.db"
  val DATABASE_TABLE = "weather"
  val WEATHER_CITY = "weather_city"
  val WEATHER_COUNTRY = "weather_country"
  val WEATHER_TEMP = "weather_temp"
  val WEATHER_TEMP_UNIT = "weather_temp_unit"
  val WEATHER_CLOUDNESS = "weather_cloudness"
  val WEATHER_DATE = "weather_date"

  val CREATE_TABLE = "create table " +
    DATABASE_TABLE + " (" +
    BaseColumns._ID + " integer primary key autoincrement, " +
    WEATHER_CITY + " text not null, " +
    WEATHER_COUNTRY + " text not null, " +
    WEATHER_TEMP + " text not null, " +
    WEATHER_TEMP_UNIT + " text not null, " +
    WEATHER_CLOUDNESS + " text not null, " +
    WEATHER_DATE + " text)"
}

class DatabaseHelper(context: Context) extends SQLiteOpenHelper(context, null, null, 1) with BaseColumns {
  import DatabaseHelper._
  override def onCreate(db: SQLiteDatabase): Unit = db.execSQL(CREATE_TABLE)
  override def onUpgrade(p1: SQLiteDatabase, p2: Int, p3: Int): Unit = ???
}

File: app/src/main/java/freemahn/com/lesson8/DatabaseHelper.java
package freemahn.com.lesson8;

import android.content.Context;
import android.database.DatabaseErrorHandler;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.provider.BaseColumns;

/**
 * Created by Freemahn on 22.01.2015.
 */
public class DatabaseHelper extends SQLiteOpenHelper implements BaseColumns {

    public static final String DATABASE_NAME = "mydatabase2.db";
    public static final String DATABASE_TABLE = "weather";
    public static final int DATABASE_VERSION = 1;
    public static final String TEXT_COLUMN = "text";
    public static final String CODE_COLUMN = "code";
    public static final String TEMP_COLUMN = "temp_current";
    public static final String TEMP_HIGH_COLUMN = "temp_high";
    public static final String DATE_COLUMN = "date";


    private static final String DATABASE_CREATE_SCRIPT = "create table "
            + DATABASE_TABLE + " (" + BaseColumns._ID
            + " integer primary key autoincrement, " + TEXT_COLUMN
            + " text, " + CODE_COLUMN + " integer, " + TEMP_COLUMN + " integer, " + DATE_COLUMN + " text not null, " + TEMP_HIGH_COLUMN + " integer); ";
    private static final String SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS "
            + DATABASE_TABLE;

    public DatabaseHelper(Context context, String name, SQLiteDatabase.CursorFactory factory,
                          int version, DatabaseErrorHandler errorHandler) {
        super(context, name, factory, version, errorHandler);
    }

    public DatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    public DatabaseHelper(Context context, String name, SQLiteDatabase.CursorFactory factory,
                          int version) {
        super(context, name, factory, version);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(DATABASE_CREATE_SCRIPT);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(SQL_DELETE_ENTRIES);
        onCreate(db);
    }
}

